pkg/cache/v2/mux.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 Envoyproxy Authors
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package cache
+
+import (
+	"context"
+	"errors"
+)
+
+type MuxCache struct {
+	// Classification functions.
+	Classify func(Request) string
+	// Muxed caches.
+	Caches map[string]Cache
+}
+
+var _ Cache = &MuxCache{}
+
+func (mux *MuxCache) CreateWatch(request Request) (chan Response, func()) {
+	key := mux.Classify(request)
+	cache, exists := mux.Caches[key]
+	if !exists {
+		value := make(chan Response, 1)
+		value <- RawResponse{Request: request}
```

Comment:
How does the receiver identify that this is a scenario where the request was not able to be classified? It feels like this is an error scenario.

Reply:
Yeah, I wasn't sure what to return. This will effectively just leave it hanging until the client decides to close it. We may need to change the interface so that the server closes the stream in some cases.

Reply:
After giving this a bit of thought, returning nil as channel works as intended. It is equivalent to a closed empty channel, so the server will close the stream.